done.js

```diff
@@ -3,241 +3,336 @@ var chroma = require('@v3rse/chroma');
+    }
 }
 
 
-
 //Lists all tasks
 function listAll() {
 
-	//data
-	var data = getData();
+    //data
+    var data = getData();
+
+    if (data.uncompleted.length || data.completed.length) {
+        // Better null checking
+        if (data.uncompleted.length) {
+            printUncompleted(data);
+        }
+        if (data.completed.length) {
+            console.log("\n");
+            printCompleted(data);
+        }
 
-	if (data.uncompleted.length || data.completed.length) {
-		
-		printUncompleted(data);
-		console.log("\n");	
-		printCompleted(data);
+    } else {
+        displayError("No tasks added!!");
+    }
 
-	} else {
-		displayError("No tasks added!!");
-	}
+}
 
+// Initalizes git integration
+function gitInit() {
+    // If we're not currently in  a git repository, exit
+    if (!fs.existsSync('.git')) {
+        displayError("Not currently in a git repository!");
+        return;
+    }
+    checkForFile('.gitignore', function () {
+        // Read in gitignore, add .database.json if not already present
+        var gitignore = fs.readFileSync('.gitignore', 'utf8');
+        if (gitignore.indexOf('.database.json') == -1) {
+            fs.appendFile('.gitignore', '\n.database.json\n', function (err) {
+                if (err) {
+                    displayError("Error modifying gitignore!");
+                    return;
+                }
+                console.log("Git integration complete");
+            });
+        }
+    });
 }
 
+function gitCommit() {
+
+    const message = process.argv[4];
+    if (!message) {
+        displayError("Git commit message required!");
+        return;
+    }
+    const task = process.argv[5] - 1;
+    if (task == undefined) {
+        displayError("Git commit message and task number required!");
+        return;
+    }
+    //get data
+    var data = getData();
+
+    if (data.uncompleted[task]) {
+        // Run git commit on escaped string
+        exec('git commit -m \"' + message.replace(/"/g, '\\"') + "\"", function (err, stdout, stderr) {
+            // If the process had an error (probably because nothing had been added with git add)
+            if (err) {
+                displayError("Git commit not possible - did you add files and include a message?");
+                return;
+            }
+            // If nothing had been added with git add
+            if (stdout.startsWith("On branch")) {
+                displayError("Git commit not possible - did you add files and include a message?");
+            }
+
+            // extract sha1 hash
+            var shaRegex = /^\[(.*) (.*)\]/gm;
+            var arr = shaRegex.exec(stdout);
+            const commitSha = arr[2];
+            //Alert the git commit has been created
+            console.log("Committed to git with SHA " + commitSha);
+            data.uncompleted[task].commit = commitSha;
+
+            //save data
+            setData(data);
+            console.log("Saved commit to task " + task + 1);
```

Comment:
I might be cool to call `check(task)` here to check the task after the commit is added.